toolkit/tools/pkg/imagecustomizerlib/imagecustomizer.go

```diff
@@ -43,6 +43,98 @@ var (
 	ToolVersion = ""
 )
 
+type ImageCustomizerParameters struct {
+	// build dirs
+	buildDir    string
+	buildDirAbs string
+
+	// input image
+	inputImageFile string
+
+	// configurations
+	configPath                  string
+	config                      *imagecustomizerapi.Config
+	customizeOSPartitions       bool
+	useBaseImageRpmRepos        bool
+	rpmsSources                 []string
+	enableShrinkFilesystems     bool
+	outputSplitPartitionsFormat string
+
+	// intermediate writeable image
+	rawImageFile string
+
+	// output image
+	outputImageFormat     string
+	qemuOutputImageFormat string
+	outputImageFile       string
+	outputImageDir        string
+	outputImageBase       string
+
+	// iso builder helper
+	isoBuilder *LiveOSIsoBuilder
+}
+
+func createImageCustomizerParameters(buildDir string,
+	inputImageFile string,
+	configPath string, config *imagecustomizerapi.Config,
+	useBaseImageRpmRepos bool, rpmsSources []string, enableShrinkFilesystems bool, outputSplitPartitionsFormat string,
+	outputImageFormat string, outputImageFile string) (*ImageCustomizerParameters, error) {
+
+	ic := &ImageCustomizerParameters{}
+
+	// working directories
+	ic.buildDir = buildDir
+
+	buildDirAbs, err := filepath.Abs(buildDir)
+	if err != nil {
+		return nil, err
+	}
+
+	ic.buildDirAbs = buildDirAbs
+
+	err = os.MkdirAll(ic.buildDirAbs, os.ModePerm)
```

Comment:
Please move the resource creation to be outside of the `createImageCustomizerParameters` function.

Reply:
pending